state/txn.go

```
@@ -187,14 +188,30 @@ func (txn *Txn) SubBalance(addr types.Address, balance *big.Int) {
 	})
 }
 
+// SubStakedBalance reduces the staked balance
+func (txn *Txn) SubStakedBalance(addr types.Address, balance *big.Int) {
+	if balance.Sign() == 0 {
+		return
+	}
+	txn.upsertAccount(addr, true, func(object *StateObject) {
+		object.Account.StakedBalance.Sub(object.Account.StakedBalance, balance)
```

Won't this move `StakedBalance` to negative if `balance` > `Account.StakedBalance`?

---

The only place this is called from is from the [unstaking handler](https://github.com/0xPolygon/polygon-sdk/pull/76/files#diff-d4e1bf18e1d1050f24fa6a0d47dc44b3ac67e2866a634b72342739164372dac9R45), where we previously check what the staked account balance was and deduct it

---

True, but I would argue that subBalance itself should return an error if this occurs. However I can see that for regular balance this isn't done as well.

---

We can always go along the route - the caller has the responsibility to ensure correct state execution

---

Ok, let's go like that for now 🙂